### DISCUSSION TITLE
gdl compile fails with error: cannot find config.h

### DISCUSSION THREAD

**Discussion by wogallery** [gnudatalanguage/gdl#1020] [Q&A]
I have tried to compile the latest gdl (on git as of 2021-07-19) in both Linux (Ubuntu) and Windows(cygwin). In both cases the build fails with the error:

[ 15%] Built target antlr
[ 16%] Building CXX object CMakeFiles/gdl.dir/CFMTLexer.cpp.o
In file included from /cygdrive/f/gdl_windows/gdl/src/CFMTLexer.cpp:3:
/cygdrive/f/gdl_windows/gdl/src/includefirst.hpp:37:10: fatal error: config.h: No such file or directory
   37 | #include <config.h>
      |          ^~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/gdl.dir/build.make:76: CMakeFiles/gdl.dir/CFMTLexer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/gdl.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

Here are the build steps:
cd ./build
cmake ../src ../build
make

Attached are the files
./src/CMakeLists.txt
./build/Makefile (renamed Makefile.txt or it would not upload)
./build/CMakeFiles/CMakeOutput.log

What am I missing? I am an idl programmer with no experience with C, C++ or cmake

Cheers,
Bill Gallery

[CMakeLists.txt](https://github.com/gnudatalanguage/gdl/files/6850861/CMakeLists.txt)
[Makefile.txt](https://github.com/gnudatalanguage/gdl/files/6850875/Makefile.txt)
[CMakeOutput.log](https://github.com/gnudatalanguage/gdl/files/6850876/CMakeOutput.log)

**Comment by slayoo** [comment]
Hello,

Please try:
`cmake ..`
instead of
`cmake ../src ..build`

HTH,
Sylwester

**Comment by alaingdl** [comment]
From my point of view, the current way with the script `.ci/build_gdl.sh` is just super simple and super robust : 
Let's start on the Linux side (the first three lines is just a bootstrap) ([tuto here](https://github.com/gnudatalanguage/gdl/wiki/GDL-on-Linux))

```
apt-get install git
apt-get install cmake
apt-get install g++
mkdir GDL
cd GDL
git clone https://github.com/gnudatalanguage/gdl
cd gdl
bash .ci/build_gdl.sh prep   <<-- need to be sudo to add set of official packages
bash .ci/build_gdl.sh build
```

Less than five minutes on most 2 or 4 cores current laptops ...

Then for the MSwin side, same script, look at [this tuto](https://github.com/gnudatalanguage/gdl/wiki/GDL-on-Windows)
